packages/snippets/bin/snippets.dart

```diff
@@ -16,23 +19,47 @@ const String _kHelpOption = 'help';
 const String _kInputOption = 'input';
 const String _kLibraryOption = 'library';
 const String _kOutputOption = 'output';
+const String _kOutputDirectoryOption = 'output-directory';
 const String _kPackageOption = 'package';
 const String _kTemplateOption = 'template';
 const String _kTypeOption = 'type';
 
 const LocalFileSystem filesystem = LocalFileSystem();
 
-String getChannelName() {
+class GitStatusFailed implements Exception {
+  GitStatusFailed(this.gitResult);
+  final ProcessResult gitResult;
+  @override
+  String toString() => 'git status exited with a non-zero exit code: ${gitResult.exitCode}:\n${gitResult.stderr}\n${gitResult.stdout}';
+}
+
+/// Get the name of the channel these docs are from.
+///
+/// First check env variable LUCI_BRANCH, then refer to the currently
+/// checked out git branch.
+String getChannelName({
+  @visibleForTesting
+  Platform platform = const LocalPlatform(),
+  @visibleForTesting
+  ProcessManager processManager = const LocalProcessManager(),
+}) {
+  final String? envReleaseChannel = platform.environment['LUCI_BRANCH']?.trim();
+  if (<String>['master', 'stable'].contains(envReleaseChannel)) {
+    return envReleaseChannel!;
+  }
   final RegExp gitBranchRegexp = RegExp(r'^## (?<branch>.*)');
-  final ProcessResult gitResult =
-      Process.runSync('git', <String>['status', '-b', '--porcelain']);
-  if (gitResult.exitCode != 0)
-    throw 'git status exit with non-zero exit code: ${gitResult.exitCode}';
-  final RegExpMatch? gitBranchMatch = gitBranchRegexp
-      .firstMatch((gitResult.stdout as String).trim().split('\n').first);
-  return gitBranchMatch == null
-      ? '<unknown>'
-      : gitBranchMatch.namedGroup('branch')!.split('...').first;
+  final ProcessResult gitResult = processManager.runSync(<String>['git', 'status', '-b', '--porcelain'],
+      environment: <String, String>{
+        'GIT_TRACE': '2',
+        'GIT_TRACE_SETUP': '2'
```

Comment:
Was this just for debugging? Or is this useful to keep?

Reply:
It's useful to keep because we were having mysterious crashes in git on CI, and this will help debug that. Added a comment.